src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -159,22 +159,10 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
         ciphertext += out_length;
     }
 
-    if (plaintext_length != 0)
```

Comment:
Unfortunately, this is not the right fix as you shouldn't assume that no output has been delayed by the last `psa_aead_update` call. Admittedly, we do `VerifyOrReturnError(partial_block_length == out_length, CHIP_ERROR_INTERNAL);` a few lines above but in fact, according to the `psa_aead_update` documentation, `psa_aead_update` called on a partial block may delay the output up until `psa_aead_finish` is called. I think the right solution should be to use `temp_buffer` for `psa_aead_finish` as well.

Also, the mentioned `VerifyOrReturnError` should be removed :)